bittorrent/request_sanitizer.go

```diff
@@ -0,0 +1,61 @@
+package bittorrent
+
+import (
+	"net"
+
+	log "github.com/sirupsen/logrus"
+)
+
+// ErrInvalidIP indicates an invalid IP for an Announce.
+var ErrInvalidIP = ClientError("invalid IP")
+
+// RequestSanitizer is used to replace unreasonable values in requests parsed
+// from a frontend into sane values.
+type RequestSanitizer struct {
+	MaxNumWant          uint32 `yaml:"max_numwant"`
+	DefaultNumWant      uint32 `yaml:"default_numwant"`
+	MaxScrapeInfoHashes uint32 `yaml:"max_scrape_infohashes"`
+}
+
+// SanitizeAnnounce enforces a max and default NumWant and coerces the peer's
+// IP address into the proper format.
+func (rs *RequestSanitizer) SanitizeAnnounce(r *AnnounceRequest) error {
```

Comment:
An idea about this:
The method doesn't need to be defined on a pointer to the struct, it could be defined on the struct itself. (Because it doesn't change it and the struct is pretty small.) And if we do that, we could also just pass it as a parameter. Then we could rename `RequestSanitizer` to something like `RequestSanitizerOptions`, which would also make the config structs further down nicer.

Thoughts?